src/backend/marsha/core/api.py

```diff
@@ -497,18 +497,34 @@ def update_live_state(
         if not validate_signature(request.headers.get("X-Marsha-Signature"), msg):
             return Response("Forbidden", status=403)
 
+        # Load the video first to return a 404 if not existing
         video = self.get_object()
-        video.live_state = serializer.validated_data["state"]
+
+        # Try to update the video with the new live state. If the video has already this live state
+        # we are in a concurrent request and only the first one should be accepted.
+        updated_rows = Video.objects.filter(
+            ~Q(live_state=serializer.validated_data["state"]),
+            pk=pk,
+        ).update(live_state=serializer.validated_data["state"])
+
+        if updated_rows == 0:
+            # State is alreay updated, we can stop the process here
+            # If we return a status different than 200 the lambda will retries
+            # to update the live state several times.
+            return Response({"success": True})
+
+        video.refresh_from_db()
+
         live_info = video.live_info
         live_info.update(
             {"cloudwatch": {"logGroupName": serializer.validated_data["logGroupName"]}}
         )
 
-        if video.live_state == defaults.RUNNING:
+        if serializer.validated_data["state"] == defaults.RUNNING:
             live_info.update({"started_at": stamp})
             video.live_info = live_info
 
-        if video.live_state == defaults.STOPPED:
+        if serializer.validated_data["state"] == defaults.STOPPED:
```

Comment:
I don't get the point of changing this since you refreshed the video from db.

Reply:
Both are good, I change them in a first time to do a test and I didn't revert this changes. I will do it.